js/controllers/authors.js

```diff
@@ -0,0 +1,69 @@
+/* global app */
+/* global angular */
+
+app.controller('authorsCtrl', ['$scope', 'authorService', 'enableEditService', 'bookService', function($scope, authorService, enableEditService, bookService, $index) {
+  
+  $scope.include = function(obj) {
+     authorService.updateAuthors(obj);
+   };
+   
+  $scope.temp = authorService.authorsList;
+  $scope.authorErrorMessage = "";
+  
+  $scope.edit = enableEditService.edit;
+  
+  $scope.gridOptions = {
+    
+    data: 'temp',
+    enableCellSelection: true,
+    enableCellEdit: true,
+    enableCellEditOnFocus: true,
+    rowEditWaitInterval: -1,
+    saveFocus: false,
+    saveScroll: true,
+    saveGroupingExpandedStates: true,
+    enableRowSelection: true,
+    enableRowHeaderSelection: false,
+    onRegisterApi: function (gridApi) {
+                   $scope.gridApi = gridApi;
+                   },
+  
+    columnDefs: [
+       { field: 'id', displayName: 'ID', enableCellEdit: false, enableHiding: false
+       },
+       { field: 'name', displayName: 'Name', enableHiding: false, cellEditableCondition: $scope.edit
+       },
+       { field: 'age', displayName: 'Age', enableHiding: false, cellEditableCondition: $scope.edit
+       }
+    ]
+  };
+  
+  $scope.addNewItem = function() {
+    var n = $scope.temp.length + 1;
+    $scope.temp.push( { id: n, name: $scope.newAuthorName, age: $scope.newAuthorAge});
+    modal.css('display', 'none');
+    };
+    
+    $scope.selected = $scope.temp[0];
```

Comment:
why does selected need to be set to the first item in the temp array?

Reply:
oops this was part of a previous solution that's been replaced, has been removed